src/Support/SupportService.php

```diff
@@ -0,0 +1,187 @@
+<?php
+/**
+ * Service class for support.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP\Support;
+
+use AmpProject\AmpWP\Infrastructure\Service;
+use AmpProject\AmpWP\Infrastructure\CliCommand;
+use AmpProject\AmpWP\Validation\ScannableURLProvider;
+use AmpProject\AmpWP\Validation\URLScanningContext;
+
+/**
+ * Class SupportService
+ * Service class for support.
+ */
+class SupportService implements Service, CliCommand {
+
+	/**
+	 * Get the name under which to register the CLI command.
+	 *
+	 * @return string The name under which to register the CLI command.
+	 */
+	public static function get_command_name() {
+
+		return 'amp';
+	}
+
+	/**
+	 * To send support data.
+	 *
+	 * @param array $args Support data argument.
+	 *
+	 * @return array|\WP_Error Response from insight server.
+	 */
+	public function send_data( $args = [] ) {
+
+		$support = new SupportData( $args );
+
+		return $support->send_data();
+	}
+
+	/**
+	 * To get support data.
+	 *
+	 * @param array $args Support data argument.
+	 *
+	 * @return array Support data.
+	 */
+	public function get_data( $args = [] ) {
+
+		$support = new SupportData( $args );
```

Comment:
The `SupportData` class should rather be passed via dependency injection in the constructor.